cypress/jenkins/slack-notification.sh

```diff
@@ -0,0 +1,134 @@
+# This script sends notifications to Slack when e2e tests fail in Jenkins
+
+set -e
+
+# Function to send Slack notification
+send_slack_notification() {
+    local status="$1"
+    local message="$2"
+    local bot_token="$3"
+    local channel="$4"
+    
+    if [ -z "$bot_token" ]; then
+        echo "Warning: UI_SLACK_BOT_TOKEN not set, skipping Slack notification"
+        return 0
+    fi
+    
+    # Prepare the JSON payload
+    local payload=$(cat <<EOF
+{
+    "channel": "$channel",
+    "text": "$message",
+    "username": "Jenkins E2E Tests"
+}
+EOF
+)
+    
+    # Send the notification using Slack Web API
+    curl -X POST \
+        -H "Content-type: application/json; charset=utf-8" \
+        -H "Authorization: Bearer $bot_token" \
+        --data "$payload" \
+        "https://slack.com/api/chat.postMessage"
+}
+
+# Function to read value from file
+read_from_file() {
+    local file_path="$1"
+    
+    if [ -f "$file_path" ]; then
+        cat "$file_path" | tr -d '[:space:]'
+    fi
+}
+
+# Function to read value from notification_values.txt
+read_notification_value() {
+    local key="$1"
+    local file_path="${WORKSPACE}/notification_values.txt"
+    
+    if [ -f "$file_path" ]; then
+        grep "^${key}=" "$file_path" | cut -d'=' -f2- | tr -d '[:space:]'
+    fi
+}
+
+# Main execution
+send_jenkins_e2e_failure_notification() {
+    local build_status="$1"
+    local job_name="${JOB_NAME:-Unknown Job}"
+    local build_number="${BUILD_NUMBER:-Unknown}"
+    local build_url="${BUILD_URL:-}"
+    
+    # Job-specific variables from init.sh
+    local rancher_version=$(read_notification_value "RANCHER_VERSION")
+    local rancher_image_tag=$(read_notification_value "RANCHER_IMAGE_TAG")
+    local rancher_chart_url=$(read_notification_value "RANCHER_CHART_URL")
+    local rancher_helm_repo=$(read_notification_value "RANCHER_HELM_REPO")
+    local cypress_tags=$(read_notification_value "CYPRESS_TAGS")
+    
+    # Get Slack bot token and channel from Secrets Manager
+    local slack_bot_token="${UI_SLACK_BOT_TOKEN:-}"
+    local slack_channel="${UI_SLACK_CHANNEL:-}"
+    
+    # Only send notifications for failures
+    if [ "$build_status" != "FAILURE" ] && [ "$build_status" != "UNSTABLE" ]; then
+        echo "Build status is $build_status, no notification needed"
+        return 0
+    fi
+
+    # Prepare the message
+    local emoji="❌"
+    local status_text="FAILED"
+    
+    if [ "$build_status" = "UNSTABLE" ]; then
+        emoji="⚠️"
+        status_text="UNSTABLE"
+    fi
+    
+    local message="*E2E Tests $status_text* $emoji\n"
+    message+="• *Job:* $job_name\n"
+    
+    # Add build number with link
+    if [ -n "$build_url" ] && [ "$build_number" != "Unknown" ]; then
+        message+="• *Build:* <$build_url|#$build_number>\n"
+    elif [ "$build_number" != "Unknown" ]; then
+        message+="• *Build:* #$build_number\n"
+    fi
+    
+    if [ "$rancher_version" != "Unknown" ]; then
```

Comment:
I suggest handling empty string values, as [ "$rancher_version" != "Unknown" ]; will be true if $rancher_version is empty. I recommend improving checks to ensure the value is not empty:

`if [ -n "$rancher_version" ] && [ "$rancher_version" != "Unknown" ]; then`
applying this for at least the rancher_image_tag, rancher_helm_repo, cypress_tags